### PR TITLE
Skip padding cost of fusible/planable inputs

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #125780
* #125773
* #125772

For mm inputs which are not inputs of the graph, assume that we can memory plan them in the aten.cat and exclude the padding cost in the benchmarking comparison. Technically we also have to do a small amount of 0s writing, but that should be relatively small and encompassed in the weighting of the padding time by `1.1` 

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/fx_passes/pad_mm.py [hunk 1]
```diff
@@ -308,20 +336,48 @@ def realize_tensor(t):
...
+        )
+        ori_time = get_cached_should_pad(ori_time_key)
+        if ori_time is None:
+            if op is torch.ops.aten.bmm or op is torch.ops.aten.mm:
+                ori_time = do_bench(
+                    lambda: op(mat1, mat2),
+                )
+            else:
+                if input is not None:
+                    input = realize_tensor(input)
```
Comment (shunting314) [inline torch/_inductor/fx_passes/pad_mm.py:353]: Explain a bit why a realize is needed here?

**Review by shunting314 — COMMENTED** [review]
Overall the pre-padding implementation looks good to me.

The main comment I have is about the cache design here.

Also unit test are failing.

File: torch/_inductor/fx_passes/pad_mm.py [hunk 2]
```diff
@@ -308,20 +336,48 @@ def realize_tensor(t):
...
+                ori_time = do_bench(
+                    lambda: op(input, mat1, mat2),
+                )
+            set_cached_should_pad(ori_time_key, ori_time)
```
[the remark this answers is not in the document]
Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:357]: Yea, i can make it a bit more explicit. The key here is different than at the end 
```
ori_time_key = "ori_time " + should_pad_bench_key(
            match, mat1, mat2, op, input, ignore_should_pad=True
        )
```

ori_time is prepended

File: torch/_inductor/fx_passes/pad_mm.py (hunk 1 above)
Comment (shunting314) [reply torch/_inductor/fx_passes/pad_mm.py:364]: hmm, I mean, why we need realize? lol

**Review by shunting314 — APPROVED** [review]

File: torch/_inductor/fx_passes/pad_mm.py (hunk 1 above)
Comment (eellison) [reply torch/_inductor/fx_passes/pad_mm.py:364]: realize just means instantiate here. I imagine `input` should never not be None here. or we should dispatch to mm. but will leave for follow up.

**Pull request closed without merging**